### DISCUSSION TITLE
Auto-generated Swagger UI error

### DISCUSSION THREAD

**Discussion by haoyuant** [aws-powertools/powertools-lambda-python#4086] [General]
Hi, I followed the docs to enable swagger and it can load my endpoints correctly, however, when I expand any endpoint it only shows an infinite loading and throws the error `Unhandled Promise Rejection: TypeError: Right side of assignment cannot be destructured`.  And the backend throws a warning `|  Invoked functions/apis/season.handler
|  +1442ms /Users/haoyuantang/Library/Caches/pypoetry/virtualenvs/functions-UNII4JFY-py3.12/lib/python3.12/site-packages/aws_lambda_powertools/event_handler/api_gateway.py:1682: UserWarning: You are using Pydantic v2, which is incompatible with OpenAPI schema 3.0. Forcing OpenAPI 3.1
|  +1442ms   spec = self.get_openapi_schema(`.

Do I miss something?

<img width="999" alt="image" src="https://github.com/aws-powertools/powertools-lambda-python/assets/45114697/ee357c71-9029-49aa-9be6-6ae821e8a5fc">

<img width="1296" alt="image" src="https://github.com/aws-powertools/powertools-lambda-python/assets/45114697/6de7a2bb-2c26-47b8-998e-7ac876e87d1b">

Here is my backend code:
```
import boto3.dynamodb
import requests
from requests import Response

from aws_lambda_powertools import Logger, Tracer
from aws_lambda_powertools.event_handler import APIGatewayHttpResolver, Response
from aws_lambda_powertools.logging import correlation_paths
from aws_lambda_powertools.utilities.typing import LambdaContext
import boto3
from aws_lambda_powertools.utilities.parser import BaseModel
from pydantic import Field, AliasPath
from aws_lambda_powertools.event_handler.openapi.params import Body  
from aws_lambda_powertools.shared.types import Annotated
import os

tracer = Tracer()
logger = Logger()
app = APIGatewayHttpResolver(enable_validation=True)
app.enable_swagger(path="/swagger")

dynamodb = boto3.resource("dynamodb")
seasonsTable = dynamodb.Table(os.environ.get("SEASONS_TABLE_NAME"))

class Season(BaseModel):
    user_id: str
    season_id: str

@app.get('/seasons')
@tracer.capture_method
def get_seasons():
    seasons = seasonsTable.scan()
    return seasons

@app.post("/seasons")
@tracer.capture_method
def create_season(season: Season):
    seasonsTable.put_item(Item={
        "userId": season.user_id,
        "seasonId": season.season_id,
    })
    return Response(status_code=200)



# You can continue to use other utilities just as before
@logger.inject_lambda_context(correlation_id_path=correlation_paths.API_GATEWAY_HTTP)
@tracer.capture_lambda_handler
def handler(event: dict, context: LambdaContext) -> dict:
    return app.resolve(event, context)
```

**Comment by kattakaha** [comment]
Hi @haoyuant 

I recommend simplifying the issue by separating it from DynamoDB. I wasn't sure what the model name "Season" represents, but I'll add my code, excluding the process of saving to DynamoDB, along with a screenshot of the actual output Swagger.

Additionally, here are the official documents you should refer to.

https://docs.powertools.aws.dev/lambda/python/latest/core/event_handler/api_gateway/#data-validation

![image](https://github.com/aws-powertools/powertools-lambda-python/assets/58053010/3d3bff6b-dd20-41f7-95a9-c3325d81b89c)

Here is my code.

```python
from typing import List

from aws_lambda_powertools.event_handler import APIGatewayRestResolver
from aws_lambda_powertools.utilities.parser import BaseModel
from aws_lambda_powertools.utilities.typing import LambdaContext



app = APIGatewayRestResolver(enable_validation=True)
app.enable_swagger(path="/swagger")


class Season(BaseModel):
    user_id: str
    season_id: str


@app.get("/seasons")
def get_seasons() -> List[Season]:
    # TODO: Replace with actual data fetch from database
    seasons = [
        Season(user_id="1", season_id="1"),
        Season(user_id="2", season_id="2"),
    ]
    return seasons


@app.post("/seasons")
def create_season(season: Season) -> Season:
    # TODO: Add logic to save season to database
    return Season(user_id=season.user_id, season_id=season.season_id)


def lambda_handler(event: dict, context: LambdaContext) -> dict:
    return app.resolve(event, context)
```

**Comment by nlykkei** [comment]
@haoyuant The OpenAPI spec used with the Swagger UI is generated based on type annotations with Pydantic models. You don't use type annotations

**Comment by leandrodamascena** [comment]
I'm closing this discussion as resolved. Please reopen if you need.
